exec/jdbc/src/test/java/org/apache/drill/jdbc/PreparedStatementTest.java

```diff
@@ -72,12 +73,18 @@
 public class PreparedStatementTest extends JdbcTestBase {
 
   private static final org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(PreparedStatementTest.class);
+  private static final Random RANDOMIZER = new Random(20150304);
 
   private static final String SYS_VERSION_SQL = "select * from sys.version";
   private static final String SYS_RANDOM_SQL =
       "SELECT cast(random() as varchar) as myStr FROM (VALUES(1)) " +
       "union SELECT cast(random() as varchar) as myStr FROM (VALUES(1)) " +
       "union SELECT cast(random() as varchar) as myStr FROM (VALUES(1)) ";
+  private static final String SYS_OPTIONS_SQL = "SELECT * FROM sys.options";
+  private static final String SYS_OPTIONS_SQL_LIMIT_10 = "SELECT * FROM sys.options LIMIT 12";
+  private static final String ALTER_SYS_OPTIONS_MAX_ROWS_LIMIT_X = "ALTER SYSTEM SET `" + ExecConstants.QUERY_MAX_ROWS + "`=";
+  //Locks used across StatementTest and PreparedStatementTest
+  private static Semaphore maxRowsSysOptionLock = StatementTest.maxRowsSysOptionLock;
```

Comment:
`connection` is created in this class, but the same `Semaphore` is used from `StatementTest` class for this test class and another one. Should they use their own instances of `Semaphore`?

Reply:
I tried that, but the unit tests attempt to test with setting up specific SYSTEM values. Without this, there is a race condition where both `StatementTest` and `PreparedStatementTest` attempt to set SYSTEM values... and get different results (resulting in test failures).

Reply:
Looks like these tests are connected to the same drillbit. In this case, the same `Semaphore` instance should be used. Thanks for the clarification.